components/raftstore/src/store/worker/pd.rs

```diff
@@ -1413,10 +1452,77 @@ fn send_destroy_peer_message<EK, ER>(
     }
 }
 
+fn collect_report_read_peer_stats(
+    capacity: usize,
+    report_read_stats: HashMap<u64, PeerReportReadStat>,
+    mut stats: pdpb::StoreStats,
+) -> pdpb::StoreStats {
+    if report_read_stats.len() < capacity * 3 {
+        for (_, read_stat) in report_read_stats {
+            let mut peer_stat = pdpb::PeerStat::default();
+            peer_stat.set_region_id(read_stat.region_id);
+            peer_stat.set_read_bytes(read_stat.read_bytes);
+            peer_stat.set_read_keys(read_stat.read_keys);
+            peer_stat.set_query_stats(read_stat.read_query_stats.0);
+            stats.peer_stats.push(peer_stat);
+        }
+        return stats;
+    }
+    let mut peer_stats = HashMap::default();
+    let mut keys_topn_report = TopN::new(capacity);
+    let mut bytes_topn_report = TopN::new(capacity);
+    let mut stats_topn_report = TopN::new(capacity);
+    for read_stat in report_read_stats.values() {
+        let mut cmp_stat = PeerCmpReadStat::default();
+        cmp_stat.region_id = read_stat.region_id;
+        let mut key_cmp_stat = cmp_stat.clone();
+        key_cmp_stat.report_stat = read_stat.read_keys;
+        keys_topn_report.push(key_cmp_stat);
+        let mut byte_cmp_stat = cmp_stat.clone();
+        byte_cmp_stat.report_stat = read_stat.read_bytes;
+        bytes_topn_report.push(byte_cmp_stat);
+        let mut query_cmp_stat = cmp_stat.clone();
+        query_cmp_stat.report_stat = read_stat.read_query_stats.get_read_query_num();
+        stats_topn_report.push(query_cmp_stat);
+    }
+
+    keys_topn_report.into_iter().for_each(|x| {
+        if let Some(report_stat) = report_read_stats.get(&x.region_id) {
+            peer_stats.insert(x.region_id, report_stat.clone());
```

Comment:
```suggestion
        if let Some(report_stat) = report_read_stats.remove(&x.region_id) {
            peer_stats.insert(x.region_id, report_stat);
```
Use `remove` instead of `get` so we don't need to cloning.